modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/sequence/steps/StepConfigTable.scala

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.seqexec.web.client.components.sequence.steps
+
+import scala.scalajs.js
+import edu.gemini.seqexec.model.Model.{Step, SystemName}
+import edu.gemini.seqexec.web.client.components.SeqexecStyles
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.component.Scala.Unmounted
+import japgolly.scalajs.react.vdom.html_<^._
+
+import scalacss._
+import scalaz.std.AllInstances._
+import scalaz.syntax.traverse._
+import scalaz.syntax.equal._
+import react.virtualized._
+
+object StepConfigTable {
+  private val CssSettings = scalacss.devOrProdDefaults
+  import CssSettings._
+
+  final case class Props(step: Step, size: Size) {
+    val settingsList: List[(SystemName, String, String)] =
+      step.config.toList.flatMap {
+        case (s, c) =>
+          c.map {
+            case (k, v) => (s, k, v)
+          }
+      }
+
+    val rowCount: Int = settingsList.size
+
+    def rowGetter(idx: Int): SettingsRow =
+      settingsList.index(idx).fold(SettingsRow.Zero)(Function.tupled(SettingsRow.apply))
+  }
+
+  // ScalaJS defined trait
+  // scalastyle:off
```

Comment:
I'm not sure what the `// scalastyle:off` and `// scalastyle:on` are there for: do they actually have meaning?

Reply:
that marks a block as don't check it by scalastyle. Otherwise it will complain about the presence of `vars`. In this case we have a Scala.js defined trait that must use `vars` to pass them to the `JS` engine